add inner/left/right/outer/semi/anti/cross-join (#2101)
JuliaData/DataFrames.jl: docs/src/man/joins.md

```

In relational database theory, this operation is generally referred to as a join.
The columns used to determine which rows should be combined during a join are called keys.

There are seven kinds of joins supported by the DataFrames package:

> -   Inner: The output contains rows for values of the key that exist in both the first (left) and second (right) arguments to `innerjoin`.

Author: Updated. I have also updated the descriptions to reflect the fact that some joins accept more than two tables.

In relational database theory, this operation is generally referred to as a join.
The columns used to determine which rows should be combined during a join are called keys.

> There are seven kinds of joins supported by the following functions in DataFrames.jl package:

Reviewer: While we're at it, this sentence was weird:
```suggestion
The following functions are provided to perform seven kinds of joins:
```
